crates/consensus/beacon/src/engine/sync.rs

```diff
@@ -345,17 +346,25 @@ where
 
 /// A wrapper type around [`SealedBlock`] that implements the [Ord] trait by block number.
 #[derive(Debug, Clone, PartialEq, Eq)]
-struct OrderedSealedBlock(SealedBlock);
+struct OrderedSealedBlock<H = Header, B = BlockBody>(SealedBlock<H, B>);
```

Comment:
let's make it accept a single `B: reth_primitives_traits::Block` generic and use `SealedBlockFor` alias internally